<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CodeEditorDialog</class>
 <widget class="QDialog" name="CodeEditorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>540</width>
    <height>497</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Code Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources_editor.qrc">
    <normaloff>:/qtgameengine/icons/script-file.png</normaloff>:/qtgameengine/icons/script-file.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QToolBar" name="toolBar">
     <property name="windowTitle">
      <string>toolBar</string>
     </property>
     <property name="movable">
      <bool>false</bool>
     </property>
     <property name="iconSize">
      <size>
       <width>16</width>
       <height>16</height>
      </size>
     </property>
     <addaction name="actionOk"/>
     <addaction name="separator"/>
     <addaction name="actionLoad"/>
     <addaction name="actionSave"/>
     <addaction name="actionPrint"/>
     <addaction name="separator"/>
     <addaction name="actionUndo"/>
     <addaction name="actionRedo"/>
     <addaction name="separator"/>
     <addaction name="actionCut"/>
     <addaction name="actionCopy"/>
     <addaction name="actionPaste"/>
     <addaction name="separator"/>
     <addaction name="actionFind"/>
     <addaction name="separator"/>
     <addaction name="actionCheck"/>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
     <item>
      <layout class="QVBoxLayout">
       <item>
        <widget class="CodeEditorWidget" name="codeEdit"/>
       </item>
       <item>
        <widget class="QStatusBar" name="statusbar">
         <property name="sizeGripEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QWidget" name="widgetSearch" native="true">
       <property name="visible">
        <bool>false</bool>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0">
        <item>
         <widget class="QGroupBox" name="groupBoxFind">
          <property name="title">
           <string>Find</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QComboBox" name="comboBoxFind">
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxCaseSensitive">
             <property name="text">
              <string>Case Sensitive</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxWholeWord">
             <property name="text">
              <string>Whole word only</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="1" column="0">
              <widget class="QPushButton" name="pushButtonFindFirst">
               <property name="text">
                <string>First</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="pushButtonFindPrev">
               <property name="text">
                <string>Previous</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="pushButtonFindNext">
               <property name="text">
                <string>Next</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="pushButtonFindLast">
               <property name="text">
                <string>Last</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxReplace">
          <property name="title">
           <string>Replace</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QComboBox" name="comboBoxReplace">
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="1" column="0">
              <widget class="QPushButton" name="pushButtonReplaceFirst">
               <property name="text">
                <string>First</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="pushButtonReplacePrevious">
               <property name="text">
                <string>Previous</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="pushButtonReplaceNext">
               <property name="text">
                <string>Next</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="pushButtonReplaceLast">
               <property name="text">
                <string>Last</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="pushButtonReplaceAll">
             <property name="text">
              <string>Replace All</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
  <action name="actionOk">
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/ok.png</normaloff>:/qtgameengine/icons/ok.png</iconset>
   </property>
   <property name="text">
    <string>OK</string>
   </property>
   <property name="toolTip">
    <string>OK, Save Changes</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/open.png</normaloff>:/qtgameengine/icons/open.png</iconset>
   </property>
   <property name="text">
    <string>Load</string>
   </property>
   <property name="toolTip">
    <string>Load the code from a text file</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/save.png</normaloff>:/qtgameengine/icons/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save the code to a text file</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/print.png</normaloff>:/qtgameengine/icons/print.png</iconset>
   </property>
   <property name="text">
    <string>Print</string>
   </property>
   <property name="toolTip">
    <string>Print the code</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/undo.png</normaloff>:/qtgameengine/icons/undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="toolTip">
    <string>Undo the last change</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/redo.png</normaloff>:/qtgameengine/icons/redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="toolTip">
    <string>Redo the last undo</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/cut.png</normaloff>:/qtgameengine/icons/cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="toolTip">
    <string>Cut the selected text into clipboard</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/copy.png</normaloff>:/qtgameengine/icons/copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="toolTip">
    <string>Copy the selected text to the clipboard</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/paste.png</normaloff>:/qtgameengine/icons/paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="toolTip">
    <string>Paste the text from the clipboard</string>
   </property>
  </action>
  <action name="actionFind">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/find.png</normaloff>:/qtgameengine/icons/find.png</iconset>
   </property>
   <property name="text">
    <string>Find</string>
   </property>
   <property name="toolTip">
    <string>Show the find and replace panel</string>
   </property>
  </action>
  <action name="actionCheck">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources_editor.qrc">
     <normaloff>:/qtgameengine/icons/check.png</normaloff>:/qtgameengine/icons/check.png</iconset>
   </property>
   <property name="text">
    <string>Check</string>
   </property>
   <property name="toolTip">
    <string>Check the script for syntax errors</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditorWidget</class>
   <extends>QPlainTextEdit</extends>
   <header>widgets/codeeditorwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources_editor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionOk</sender>
   <signal>triggered()</signal>
   <receiver>CodeEditorDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionUndo</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRedo</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>codeEdit</sender>
   <signal>undoAvailable(bool)</signal>
   <receiver>actionUndo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>codeEdit</sender>
   <signal>redoAvailable(bool)</signal>
   <receiver>actionRedo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCut</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCopy</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPaste</sender>
   <signal>triggered()</signal>
   <receiver>codeEdit</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>codeEdit</sender>
   <signal>copyAvailable(bool)</signal>
   <receiver>actionCut</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>codeEdit</sender>
   <signal>copyAvailable(bool)</signal>
   <receiver>actionCopy</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFind</sender>
   <signal>toggled(bool)</signal>
   <receiver>widgetSearch</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
